### DISCUSSION TITLE
Gridding RHI data

### DISCUSSION THREAD

**Discussion by wxwolff** [ARM-DOE/pyart#1323] [Q&A]
Is it possible to grid or generate an xarray dataset an RHI file? I am looking to be able to use the vertically high-res RHI data to extract columns along the ray.

**Comment by zssherman** [comment]
Hi @wxwolff , in the grid object there is a method to_xarray(). After gridding your rhi file with grid_from_radars you can use grid.to_xarray() which will then convert the grid to a xarray dataset. For converting straight to an xarray dataset for the rhi, @mgrover1, do we have an example for that?

**Comment by wxwolff** [comment]
Thanks. That’s what I was thinking, but have not been successful setting up the grid given only a single or just a few azimuths.

D

_____________________________________
David B. Wolff, Code 612
GPM Deputy Project Scientist for GV
GPM Ground Validation System Manager
NASA Wallops Flight Facility
34200 Fulton Street
Building N159, Room E220
Wallops Island, VA 23337
Ph: 757-824-1515; Cell: 443-262-1146


From: Zach Sherman ***@***.***>
Reply-To: ARM-DOE/pyart ***@***.***>
Date: Wednesday, November 9, 2022 at 11:29 AM
To: ARM-DOE/pyart ***@***.***>
Cc: "Wolff, David B. (WFF-6120)" ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [ARM-DOE/pyart] Gridding RHI data (Discussion #1323)


Hi @wxwolff<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fwxwolff&data=05%7C01%7CDavid.B.Wolff%40nasa.gov%7Ce8904d07c7264e8e8d4908dac26f9c05%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C638036081854544264%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=zYqurlY2D0VwsQonk6OqD3oWG57B0gnYk8u8nu8hyX8%3D&reserved=0> , in the grid object there is a method to_xarray(). After gridding your rhi file with grid_from_radars you can use grid.to_xarray() which will then convert the grid to a xarray dataset. For converting straight to an xarray dataset for the rhi, @mgrover1<https://gcc02.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmgrover1&data=05%7C01%7CDavid.B.Wolff%40nasa.gov%7Ce8904d07c7264e8e8d4908dac26f9c05%7C7005d45845be48ae8140d43da96dd17b%7C0%7C0%7C638036081854700476%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=WE9nruQBvKaAJQOD7YsqtxBeYCKfu%2BkN%2Fq%2BVjYkNtxM%3D&reserved=0>, do we have an example for that?

**Reply by mgrover1** [reply]
Do you have a sample file? Which format is it in?

**Comment by wxwolff** [comment]
Sure, thanks. I compressed it for email. It is from NASA’s NPOL radar and is in cfRadial format.



_____________________________________
David B. Wolff, Code 612
GPM Deputy Project Scientist for GV
GPM Ground Validation System Manager
NASA Wallops Flight Facility
34200 Fulton Street
Building N159, Room E220
Wallops Island, VA 23337
Ph: 757-824-1515; Cell: 443-262-1146


From: Max Grover ***@***.***>
Reply-To: ARM-DOE/pyart ***@***.***>
Date: Wednesday, November 9, 2022 at 12:30 PM
To: ARM-DOE/pyart ***@***.***>
Cc: "Wolff, David B. (WFF-6120)" ***@***.***>, Mention ***@***.***>
Subject: [EXTERNAL] Re: [ARM-DOE/pyart] Gridding RHI data (Discussion #1323)


Do you have a sample file? Which format is it in?

**Reply by mgrover1** [reply]
Can you upload it here, or email to mgrover@anl.gov?
